src/thor/timedistancebssmatrix.cc

```diff
@@ -538,12 +535,16 @@ bool TimeDistanceBSSMatrix::UpdateDestinations(
 }
 
 // Form the time, distance matrix from the destinations list
-std::vector<TimeDistance> TimeDistanceBSSMatrix::FormTimeDistanceMatrix() {
-  std::vector<TimeDistance> td;
+void TimeDistanceBSSMatrix::FormTimeDistanceMatrix(Api& request) {
+  uint32_t idx = 0;
   for (auto& dest : destinations_) {
-    td.emplace_back(dest.best_cost.secs, dest.distance);
+    Matrix::TimeDistance& td = *request.mutable_matrix()->mutable_time_distances()->Add();
+    td.set_from_index(idx / static_cast<uint32_t>(request.options().targets().size()));
+    td.set_to_index(idx % static_cast<uint32_t>(request.options().targets().size()));
+    td.set_distance(dest.distance);
+    td.set_time(dest.best_cost.secs);
+    idx++;
   }
```

Comment:
could also turn this into a templated function in `matrix_common.h` for all matrix algos to share